src/en/coloredmanga/src/eu/kanade/tachiyomi/extension/en/coloredmanga/ColoredManga.kt

```diff
@@ -0,0 +1,417 @@
+
+    // Details
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return getManga(manga.url)
+    }
+
+    override fun getMangaUrl(manga: SManga): String = "$baseUrl${manga.url}"
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val manga = response.parseAs<Mangas>()
+        return SManga.create().apply {
+            title = manga.name
+            url = "/manga/${manga.id}"
+            author = manga.author
+            artist = manga.artist
+            genre = manga.tags.joinToString()
+            description = manga.synopsis
+            status = when (manga.status.lowercase()) {
+                "Ongoing" -> SManga.ONGOING
+                "Cancelled" -> SManga.CANCELLED
+                "Hiatus" -> SManga.ON_HIATUS
+                else -> COMPLETED
+            }
+            thumbnail_url = "$baseUrl${manga.cover}"
+            initialized = true
+        }
+    }
+
+    // Chapters
+
+    override fun chapterListRequest(manga: SManga): Request = mangaDetailsRequest(manga)
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val manga = response.parseAs<Mangas>()
+
+        val listMangas = if (manga.chapters.isNotEmpty()) {
+            manga.chapters.map { chapter ->
+                SChapter.create().apply {
+                    name = listOf(chapter.number, chapter.title)
+                        .joinToString(" - ")
+                        .trim()
+                        .replace(nameRegex, "")
+                        .replace(chapterNameRegex, " $1")
+
+                    url = "/manga/${manga.id}/${chapter.number}"
+                    date_upload = try {
+                        chapterDateFormat.parse(chapter.date)!!.time
+                    } catch (e: Exception) {
+                        0L
+                    }
+                }
+            }
+        } else {
+            manga.volume.flatMap { volume ->
+                volume.chapters.map { chapter ->
+                    SChapter.create().apply {
+                        name = "${volume.number} | ${chapter.number} - ${chapter.title}".replace(nameRegex, "").replace(chapterNameRegex, " $1")
+                        url = "/manga/${manga.id}/${chapter.number}"
+                        date_upload = try {
+                            chapterDateFormat.parse(chapter.date)!!.time
+                        } catch (e: Exception) {
+                            0L
+                        }
+                    }
+                }
+            }
+        }
+
+        return listMangas.reversed()
+    }
+
+    override fun getFilterList(): FilterList = getFilters()
+
+    // Pages
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        return getManga(chapter.url.substringBeforeLast("/"))
+    }
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {

```

Comment:
similarly here, you don't need to override fetchPageList

Reply:
Hello, is there a way to get the chapter.url in `pageListParse`?

Reply:
you can pass it in url fragment... but it's fine, I didn't notice chapter.url usage previosuly